src/main.rs

```diff
@@ -54,23 +54,14 @@ fn main() {
         .unwrap_or_else(|_| "/".into());
 
     let args: Vec<String> = env::args().collect();
-    let config = match Config::new(&args, root) {
-        Ok(ok) => ok,
-        Err(e) => {
-            println!("{}", e);
-            std::process::exit(1);
-        }
-    };
+    let config = Config::new(&args, root)?;
 
     if config.devices.is_empty() {
         println!("No devices configured, exiting.");
-        std::process::exit(0);
+        return Ok(());
     }
 
-    if let Err(e) = run(config) {
-        println!("{}", e);
-        std::process::exit(2);
```

Comment:
It is used to distinguish failure type in scripts. 1 == runtime failure, 2 == configuration error, etc.

Reply:
shall I keep it that way or you don't have much preference here?

Reply:
I don't think this was used for anything. So if this return code stands in the way of a nice refactoring, then we can drop it.